File: src/prepare_census.mjs
import * as csv from 'csv/sync';
import storage from '@google-cloud/storage';
import functions from '@google-cloud/functions-framework';

functions.http('prepare_data', async (req, res) => {
  const client = new storage.Storage();
  const rawBucket = client.bucket('mjumbewu_musa_509_raw_data');
  const processedBucket = client.bucket('mjumbewu_musa_509_processed_data');

  const rawBlob = rawBucket.file('census/census_population_2020.json');
  const [content] = await rawBlob.download();
  const data = JSON.parse(content);

  const processedBlob = processedBucket.file('census_population_2020/data.csv');
  const outContent = csv.stringify(data);
  await processedBlob.save(outContent, { resumable: false });

  res.status(200).send('OK');
});

File: src/prepare_census/prepare_census.py
import csv
import io
import json
from google.cloud import storage
import functions_framework

@functions_framework.http
def prepare_data(request):
    client = storage.Client()
    raw_bucket = client.bucket('mjumbewu_musa_509_raw_data')
    processed_bucket = client.bucket('mjumbewu_musa_509_processed_data')

    raw_blob = raw_bucket.blob('census/census_population_2020.json')
    content = raw_blob.download_as_string()
    data = json.loads(content)

    processed_blob = processed_bucket.blob('census_population_2020/data.csv')
    outfile = io.StringIO()
    writer = csv.writer(outfile)
    writer.writerows(data)
    processed_blob.upload_from_string(outfile.getvalue(), content_type='text/csv')

    return 'OK'
